### PR TITLE
Create 0441-arranging-coins.js

### PR DESCRIPTION
**Pull request by aadil42**
Solved 0441-arranging-coins.js in JS.

File(s) Added: 0441-arranging-coins.js
Language(s) Used: JavaScript
Submission URLs: https://leetcode.com/problems/arranging-coins/submissions/977117480/ (linear)
                               https://leetcode.com/problems/arranging-coins/submissions/977117662/ (binary search)
                               https://leetcode.com/problems/arranging-coins/submissions/984649285/ (math)

### CODE REVIEW

File: javascript/0441-arranging-coins.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  
+/** 
+ * Math 
+ * Time O(1) | Space O(1)
+ * @param {number} n
+ * @return {number}
+ */
+var arrangeCoins = function(n) {
+    
+    let result1 = Math.floor((-1 + Math.sqrt(1+(8*n)))/2);
```
Comment (aakhtar3) [inline javascript/0441-arranging-coins.js:58]: Suggestion: Make the math operations more readable

```js
const value = Math.sqrt(1 + (8 * n)))
const add = ((value + -1) >> 1)
const sub = ((value - -1) >> 1)
```

Comment (aadil42) [reply javascript/0441-arranging-coins.js]: Done. Please check.

**Comment by aadil42** [comment]
@aakhtar3, any update on this? I don't get it, what do you mean by the period?

**Review by aakhtar3 — CHANGES_REQUESTED** [review]

File: javascript/0441-arranging-coins.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        steps++;
+    }
+
+    return canBuild || 1;
+};
+
+/**
+ * Binary Search
+ * 
+ * Time O(n*log(n)) | Space O
```
[the remark this answers is not in the document]
Comment (aadil42) [reply javascript/0441-arranging-coins.js:25]: I added the Space complexity. Sorry for that. I have a question, in the binary-search solution, I tried to get the middle value with bit-operation. It is giving me the wrong answer for this input (n=2147483647). Any idea? I checked the number, it's (2^31 - 1).  And we're adding 1 to that number. We're overflowing out of the bits. To fix this I can add the below condition
```javascript```
    if(n === 2**31 - 1) {
        right = right - 1;
    }
``` 
It does work [here](https://leetcode.com/problems/arranging-coins/submissions/1042832976/) 
I know in other binary-search questions we did bit manipulation to get mid value. All the solutions were submitted successfully, but can they cause any trouble in the future? If Leetcode decides to change the constraint to those questions?

[the remark this answers is not in the document]
Comment (aadil42) [reply javascript/0441-arranging-coins.js:25]: Let me know if I didn't explain myself clearly. Sorry for the long ass question.

File: javascript/0441-arranging-coins.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        canBuild++;
+        steps++;
+    }
+
+    return canBuild || 1;
+};
+
+/**
+ * Binary Search
+ * Time O(n*log(n)) | Space O(1)
```
Comment (aakhtar3) [inline javascript/0441-arranging-coins.js:24]: Should be log(n) for binary search

**Review by aakhtar3 — CHANGES_REQUESTED** [review]

File: javascript/0441-arranging-coins.js (hunk 3 above)
Comment (aadil42) [reply javascript/0441-arranging-coins.js]: Sorry 😅. I updated it.

**Pull request merged**